contract/src/Custody.sol

```diff
@@ -27,459 +28,276 @@ contract Custody is IChannel, IDeposit {
     error InvalidAmount();
     error TransferFailed(address token, address to, uint256 amount);
     error ChallengeNotExpired();
-    error ChannelNotFinal();
     error InsufficientBalance(uint256 available, uint256 required);
 
-    // Index in the array of participants
-    uint256 constant HOST = 0;
-    uint256 constant GUEST = 1;
-
     // Recommended structure to keep track of states
     struct Metadata {
-        Channel chan; // Opener define channel configuration
-        Status status; // Current channel status
-        uint256 challengeExpire; // If non-zero channel will resolve to lastValidState when challenge Expires
-        State lastValidState; // Last valid state when adjudicator was called
-        mapping(address token => uint256 balance) tokenBalances; // Token balances for the channel
+      Channel chan; // Opener define channel configuration
+      Status stage;
+      address creator;
+      Amount[] expectedDeposits; // Creator defines Token per participant
+      Amount[] actualDeposits; // Tracks deposits made by each participant
```

Comment:
This can be simplified by counting "processed/joined participants" instead of funds, thus using `EnumerableSet` instead of these 2 arrays (please, let's not use vanilla Solidity arrays at least).

Reply:
I didnt read yet source code of Enumerable, Im not confortable using it, we have strict index position, we can prealloc the whole array when we create channel with zeros and all access by index its performant.

Im not yet in favor of using too much libs, I prefer vanilla, we can deal with this later

Reply:
Deal later it is. Not resolving this comment to revert back to it.